mlir/lib/Analysis/DataFlowFramework.cpp

```diff
@@ -104,12 +104,16 @@ Location LatticeAnchor::getLoc() const {
 //===----------------------------------------------------------------------===//
 
 LogicalResult DataFlowSolver::initializeAndRun(Operation *top) {
+  // Enable enqueue to the worklist.
+  isRunning = true;
```

Comment:
You can use `llvm::make_scope_exit` to ensure the flag is reset upon leaving the function